README.md

```diff
@@ -86,6 +86,67 @@ def play(self):
 			return self.energy
 ```
 
+and add the following lines right under `time.sleep(self.latency)` on line 27:
+
+```python
+if debug:
+	next = input()
+	if next:
+		return 'Next'
+```
+
+### Edit `Map.py`
+Lastly, in `Map.py`, add:
+
+```python
+self.overlap = [str(i) for i in range(10)]
+```
+
+on line 19, right below:
+
+```python
+self.botPosition = self.checkpoint[:] #bot always starts at the checkpoint` on line 18.
+```
+
+Then, change the signature of the `moveRobot()` method on line 58, from:
+
+```python
+def moveRobot(self, move)
+```
+
+to:
+
+```python
+def moveRobot(self, move, debug=False)
+```
+
+and change the lines after:
+
+```python
+elif move == 'right':
+	self.botPosition[1] = self.botPosition[1]+1
+	if self.map[self.botPosition[0]][self.botPosition[1]] == '@':
+		self.remainingStains -=1
+```
+
+to the ones below, but make sure to stay within the `moveRobot()` method:
+
+```python
+self.currentSign = self.map[self.botPosition[0]][self.botPosition[1]]
+self.map[self.botPosition[0]][self.botPosition[1]] = 'B'
+if currentPosition == self.checkpoint:
+	self.map[currentPosition[0]][currentPosition[1]] = '1'
+else:
+	if self.currentSign in self.overlap:
+		if self.currentSign == '9' or self.currentSign == 'S':
+			self.overlap.append('S')
```

Comment:
If we already have a `S` in `self.overlap`, we don't need to append another one afterwards. This happened several times when I was testing with randomBot.

One possible solution may be appending 'S' to `self.overlap` right after it has been initialized. This way, the operation will only be executed once.
```python
self.overlap = [str(i) for i in range(10)]
self.overlap.append('S')
```

Reply:
Agree!

Reply:
Changes have been committed in [82c7691](https://github.com/Celqaz/beatBruteBot/pull/1/commits/82c76910c0ebef8972a09cc1f17c8ca532bfc2b2); conversition resolved.